hy/core/language.hy

```diff
@@ -369,12 +369,15 @@
   "Chunks coll into n-tuples (pairs by default). The remainder, if any, is not
    included unless a fillvalue is specified. The step defaults to n, but can be
    more to skip elements, or less for a sliding window with overlap."
-  (setv
-   step (or step n)
-   slices (genexpr (itertools.islice coll start None step) [start (range n)]))
-  (if (is fillvalue -sentinel)
-    (apply zip slices)
-    (apply zip-longest slices {"fillvalue" fillvalue})))
+  (if (< n 1) [] (do
```

Comment:
Huh. I'm not sure about this part. This kind of feels like a pointless check. It already handles `n == 0` properly without this. The `n < 0` case doesn't really make sense, so it probably should be "undefined" for this function. I know it was the empty list before, but throwing an exception might actually be better. I compared it to Clojure, which also returns `()` for the `n < 0` case, so there's that.

On the other hand, Clojure actually _diverges_ in the `n == 0` case, which is not good behavior all. This leads me to believe that all `n<1` inputs are nonsensical and should be an error.